.github/ISSUE_TEMPLATE/bug.yml

```diff
@@ -11,6 +11,9 @@ body:
       label: "`brew config` output"
     validations:
       required: true
+  - type: markdown
+    attributes:
+      value: If you are using `HOMEBREW_BOTTLE_DOMAIN` or `HOMEBREW_ARTIFACT_DOMAIN` to configure Homebrew mirrors, please first try unsetting those variables. If this bug is related to mirror sites themselves, please report the bug to the respective mirror sites. See https://github.com/Homebrew/discussions/discussions/1917 for more information.
   - type: textarea
```

Comment:
I wonder if it would be better to request unsetting `HOMEBREW_{BOTTLE,ARTIFACT}_DOMAIN` and `HOMEBREW_{BREW,CORE}_GIT_REMOTE` as part of the checkboxes below instead.

Reply:
Agree

Reply:
It might also be nice for the error message from `brew` to refer users to the right place for network failures that occur when they have these variables set.

Not sure if there's a way to do this without also carrying around a hash of remotes and associated issue trackers in `brew`, which I don't think we'll want to do (we don't know how to keep such a list up-to-date), but just mentioning this in case you have ideas here.

Reply:
I have added a checkbox and moved instructions for reporting after the checkbox section.

In fact, the failure of mirror sites does not necessarily lead to network issues. There might be
* HTTP redirections, users are simply redirected to the upstream and eventually succeed.
* Slow download speed, and users may cancel installation without seeing the error message.

These are all possible failures of mirror sites without a error reported in the console, and I believe we should let them know this when they want to report issues to Homebrew.

Reply:
> It might also be nice for the error message from `brew` to refer users to the right place for network failures that occur when they have these variables set.
> 
> Not sure if there's a way to do this without also carrying around a hash of remotes and associated issue trackers in `brew`, which I don't think we'll want to do (we don't know how to keep such a list up-to-date), but just mentioning this in case you have ideas here.

We already have a message that says this for third-party tap issues: 

```
Please report this issue to the rylan12/personal tap (not Homebrew/brew or Homebrew/core), or even better, submit a PR to fix it:
  /usr/local/Homebrew/Library/Taps/rylan12/homebrew-personal/Formula/foo.rb:16
```

I could definitely see us catching certain errors and saying something like this if one of the env vars is set:

```
Please report this issue to the maintainers of <MIRROR> mirror. For help using mirrors, see https://github.com/Homebrew/discussions/discussions/1917
```

That way, no need for a hash for each mirror. Just redirect them to that discussion (which we can structure to try to lead folks to the correct mirror repo (or but reporting location).

And good to have this in here (i.e. this PR), too, in case people ignore that.

Reply:
I've taken a quick look at the brew source code (I'm new to Ruby 😂😂), and I don't know where to add the error message for mirrors. I was thinking that I could add a new branch around where `Please report this issue to...` is printed out, but this message only shows in `check_conflict` in `formula_installer.rb` and `odeprecated` in `utils.rb`. 

I'm expecting to find some places to handle download errors and add our error messages. I'm also thinking of adding some warnings in `brew doctor` command. I would do this in another PR. What do you think?

Reply:
`brew doctor` warnings are probably a good idea. No worries if you can't get the error message thing. It might be very difficult to detect which errors are related to mirrors versus the main brew code, anyway (tbh I'm not sure what types of errors we would want to catch anyway). If you do get it, a separate PR is totally fine.